### PR TITLE
Add Cypress Testing Library

### PR DESCRIPTION
**Pull request by weeman1337**
Task from one of our recent weeklies.

- Add [Cypress Testing Library](https://github.com/testing-library/cypress-testing-library)
- Add some convenience commands like `findButton(name)`
- Implement an example test with it
- Update doc

PSF-1968

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by justjanne — APPROVED** [review]

**Comment by luixxiul** [comment]
@weeman1337 I opened a [PR](https://github.com/matrix-org/matrix-react-sdk/pull/10471) by using the library (the library looks great!), and I am just wondering if it could have more than one argument, such as: [`.findButton("Options", { timeout: 1000 })`](https://github.com/matrix-org/matrix-react-sdk/pull/10471/files#diff-8aa427661b195fe975f24c7fcc49a73653161921f730b265699f3492821bebd6R401). The timeout option has been added since sometimes Cypress did not get/click the button if hovering occurs too fast. Instead, does `findByRole` wait automatically until the button becomes interactive? Thanks in advance.

**Comment by weeman1337** [comment]
PR welcome :) I would suggest to just pass all options through to `findByRole`.

File: cypress/support/find.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    return cy.findByRole("textbox", { name });
+});
+
+Cypress.Commands.add("findOption", (name: string): Chainable<JQuery> => {
+    return cy.findByRole("option", { name });
+});
+
+Cypress.Commands.add("findMenuitem", (name: string): Chainable<JQuery> => {
+    return cy.findByRole("menuitem", { name });
+});
```
Comment (richvdh) [inline cypress/support/find.ts:67]: @weeman1337 I think I'm missing the value of these helper functions.

As someone relatively new to the react-sdk codebase and cypress, and completely new to the cypress testing library, I spent a while trying to find documentation online these helper functions: I didn't realise that they were helpers that we had defined ourselves.

Given they are so short, why not just inline them? 

In other words: I feel like these helpers are making our codebase less accessible to new contributors, whilst doing little to simplify the code.

Comment (weeman1337) [reply cypress/support/find.ts:67]: :shrug: I don't have a strong opinion on them. My intention was that „I want to find a button“ by `findButton("Submit")` is probably better understandable than `findByRole(`button`, { name: "Submit" })`.

@richvdh would you like to raise this in one of our discussions? We can then decide whether to inline and remove them.

Comment (richvdh) [reply cypress/support/find.ts:67]: Just for the record: we are going to remove them; it's cleaner and more flexible just to call `cy.findByRole` directly.

**Pull request merged**